### PR TITLE
Add 14.XXX DPT type

### PR DESCRIPTION
**Pull request by bozzo**
Add datapoint class 14 and minor improvements.

### CODE REVIEW

File: knx/dpt/formats.go [hunk 1]
```diff
@@ -4,11 +4,13 @@
...
 
 import (
+	"encoding/binary"
 	"errors"
+	"math"
 )
 
 // ErrInvalidLength is returned when the application data has unexpected length.
-var ErrInvalidLength = errors.New("Given application data has invalid length")
+var ErrInvalidLength = errors.New("given application data has invalid length")
```
Comment (vapourismo) [inline knx/dpt/formats.go:13]: Why?

File: knx/dpt/formats.go [hunk 2]
```diff
@@ -92,20 +108,15 @@ func unpackU8(data []byte, i *uint8) error {
 
 func packU32(i uint32) []byte {
 	buffer := []byte{0, 0, 0, 0, 0}
-	buffer[1] = uint8(i >> 24)
-	buffer[2] = uint8(i >> 16)
-	buffer[3] = uint8(i >> 8)
-	buffer[4] = uint8(i & 0xff)
-	return buffer
+	binary.BigEndian.PutUint32(buffer[1:], i)
+	return buffer[:]
```
Comment (vapourismo) [inline knx/dpt/formats.go:112]: What's the benefit of copying the buffer before returning?

File: knx/dpt/formats.go (hunk 1 above)
Comment (bozzo) [reply knx/dpt/formats.go:13]: My IDE considers this as a code smell issue regarding the error convention, i can revert it if you prefer?

File: knx/dpt/formats.go (hunk 2 above)
Comment (bozzo) [reply knx/dpt/formats.go]: In that case, none. It was a copy / paste from the code of another application. I revert it.

File: knx/dpt/formats.go (hunk 1 above)
Comment (vapourismo) [reply knx/dpt/formats.go:13]: Keep it, it is fine. I was just not expecting it given it is completely unrelated to the change.

**Review by vapourismo — APPROVED** [review]

**Comment by vapourismo** [comment]
Thanks for contributing this :heart:

**Pull request merged**